### PR TITLE
refactor: use `requestAnimationFrame` to replace `readTask`

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #10348

## Summary

We've been using `readTask` in a few places to ensure proper timing for DOM reads w/o triggering layout thrashing. Looking at [the implementation](https://github.com/ionic-team/stencil/blob/7ede77a873486b5cf47f0b26571f852675f67dd6/src/client/client-task-queue.ts#L23), we might be able to get close or similar behavior by using `requestAnimationFrame`.

### CODE REVIEW

File: packages/calcite-components/src/utils/openCloseComponent.ts [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import { readTask } from "@stencil/core";
 import { whenTransitionDone } from "./dom";
 
 /**
  * Exported for testing purposes only
  */
-export const internalReadTask = readTask;
+export const internalRaf = requestAnimationFrame;
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/openCloseComponent.ts:6]: I'll try to remove this before installing.

File: packages/calcite-components/src/utils/openCloseComponent.spec.ts [hunk 2]
```diff
@@ -9,7 +9,10 @@ describe("openCloseComponent", () => {
     let dispatchTransitionEvent: TransitionEventDispatcher;
 
     beforeEach(() => {
-      jest.spyOn(openCloseComponent, "internalReadTask").mockImplementation((task) => task(1337));
+      jest.spyOn(openCloseComponent, "internalRaf").mockImplementation((task) => {
+        task(1337);
+        return 1337;
```
Comment (driskull) [inline packages/calcite-components/src/utils/openCloseComponent.spec.ts:14]: 1337!

File: packages/calcite-components/src/utils/openCloseComponent.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/utils/openCloseComponent.ts:6]: Yeah seems like this just needs to be cleaned up to just use requestAnimationFrame by itself.

**Review by driskull — APPROVED** [review]
👍

**Comment by jcfranco** [comment]
@nel11211 Let us know if you are able to narrow it down to this.

**Comment by nel11211** [comment]
> @nel11211 Let us know if you are able to narrow it down to this.

Ok I think I see what's happening. `arcgis-tab-manager` triggers a dynamic `click()` [here](https://devtopia.esri.com/WebGIS/arcgis-web-analysis/blob/master/packages/analysis-studio-components/src/components/arcgis-tabs-manager/arcgis-tabs-manager.tsx#L367) when a tab `select()` function is called. When the dropdown is opened, Charts also opens a new tab for the chart in question and selects it. The dynamic click is being interpreted by the `calcite-dropdown` as a click outside the dropdown, triggering an auto-close.

@kumarGayu Is a dynamic click necessary for the function that selects a tab?

**Comment by nel11211** [comment]
Opened an issue in Analysis repo if we want to move discussion there: https://devtopia.esri.com/WebGIS/arcgis-web-analysis/issues/7297

**Comment by nel11211** [comment]
> @nel11211 Thanks for sharing your findings! If I’m reading this right, this doesn’t seem to be related to this PR. Would you mind moving the conversation to a more appropriate place?

It's not the root cause no, but worth noting for some reason the issue wasn't happening until the change from this PR.

**Pull request merged**